Cargo.toml

```diff
@@ -11,17 +11,69 @@ categories = ["embedded", "no-std", "network-programming"]
 readme = "README.md"
 
 [dependencies]
-sha1 = "0.6"
-heapless = "0.5"
-byteorder = { version = "1.4", default-features = false }
-httparse = { version = "1.4", default-features = false }
-rand_core = "0.6"
+async-std = { version = "1.12.0", optional = true }
+base64 = { version = "0.13.0", default-features = false }
+base64-simd = { version = "0.5.0", default-features = false, optional = true }
+byteorder = { version = "1.4.3", default-features = false }
+cfg-if = "1.0.0"
+core2 = "0.4.0"
+futures = { version = "0.3.21", optional = true }
+heapless = "0.7.14"
+httparse = { version = "1.7.1", default-features = false }
+rand_core = "0.6.3"
+sha1 = "0.10.1"
+smol = { version = "1.2.5", optional = true }
+tokio = { version = "1.19.2", features = ["io-util"], optional = true }
```

Comment:
At this point in time I do not wish to add tokio, smol, async-std and all their baggage to this codebase. I do plan on adding async support but I wanted to think about it properly before attempting it. I think the async streaming story in Rust is a tire fire and I don't want to add to it.

Reply:
ok, we should keep async feature then

Reply:
and then for each stream the user will need to provide their own compat implementation

Reply:
Yes I think that would be better